test/services/database/comment_repository_test.dart

```diff
@@ -248,6 +248,26 @@ void main() {
 
         expect(post.data.commentCount, equals(5));
       });
+
+      test("can delete all comments", () async {
```

Comment:
You are not testing that the comment upvotes are deleted too (which should be a failing test here I think). To do this, you will need acess to the comment upvote repository service, so I'd suggest the following refactor: you can create a function outputting the repository service corresponding to some post id in `comment_upvote_repository_service.dart`, and use this function in the provider. You can then use this function in this test.
I recommend you contacting @yoannLafore to discuss this, as he may also implement need/implement this for his tasks.

Edit: After some reflection, I am realising that, instead of a method creating the `UpvoteRepository` it might be interesting to simply create a new class (to better match the other repository interface):
```
class CommentUpvoteRepositoryService
    extends UpvoteRepositoryService<CommentIdFirestore> {
  CommentUpvoteRepositoryService({
    required super.firestore,
    required PostIdFirestore postId,
  }) : super(
          parentCollection: firestore
              .collection(PostFirestore.collectionName)
              .doc(postId.value)
              .collection(CommentFirestore.subCollectionName),
          voteScoreField: CommentData.voteScoreField,
        );
}

final commentUpvoteRepositoryProvider = Provider.family<
    UpvoteRepositoryService<CommentIdFirestore>,
    PostIdFirestore>((ref, postId) {
  return CommentUpvoteRepositoryService(
    firestore: ref.watch(firestoreProvider),
    postId: postId,
  );
});
```
Some documentation would probably need to be added to this class, naturally.

Reply:
The specialized providers exist, I don't think its for this pr to decide whether we need specialized classes as well.

I added a test to the single comment deletion that checks that upvotes are deleted as well. 30561afd5e78ed75a66439b524df03c611a968e3

Reply:
After more discussion, with @yoannLafore as well, we decided this was a good idea but it was easier to do by making the current constructor of `UpvoteRepositoryService` private and make simpler public constructors that do not rely on the database structure. This was done in #237.

I integrated this refactor into this issue in 5a43590d69a679cbfa0db0e06800e3da40470ddf

Reply:
Thank you :)